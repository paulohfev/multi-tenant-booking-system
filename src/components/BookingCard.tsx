import { Booking } from '@/types/Booking'
import { formatDate } from '@/utils/date'

export type BookingCardProps = {
  booking: Booking
}

const BookingCard: React.FC<BookingCardProps> = ({ booking }) => {
  return (
    <div className='bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200'>
      <h3 className='text-xl font-semibold text-gray-900 mb-2'>{booking.serviceName}</h3>

      <p className='text-gray-600 mb-2'>{booking.name}</p>
      <p className='text-gray-600 mb-2'>{booking.email}</p>

      <footer className='flex justify-between items-center mt-4'>
        <p className='text-gray-600'>{formatDate(booking.date)}</p>
        <p className='text-gray-600'>{booking.time}</p>
      </footer>
    </div>
  )
}

export default BookingCard
