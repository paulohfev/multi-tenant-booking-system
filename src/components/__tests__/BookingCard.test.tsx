import { render } from '@testing-library/react'
import BookingCard from '../BookingCard'

const mockBooking = {
  id: '1',
  name: 'John Doe',
  email: 'john@example.com',
  date: '2024-03-21',
  time: '14:30',
  serviceName: 'Test Service',
}

describe('BookingCard', () => {
  it('should match snapshot', () => {
    const { container } = render(<BookingCard booking={mockBooking} />)
    expect(container).toMatchSnapshot()
  })
})
