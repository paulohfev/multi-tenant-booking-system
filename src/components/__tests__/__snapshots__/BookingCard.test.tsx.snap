// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCard should match snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200"
  >
    <h3
      class="text-xl font-semibold text-gray-900 mb-2"
    >
      Test Service
    </h3>
    <p
      class="text-gray-600 mb-2"
    >
      John Doe
    </p>
    <p
      class="text-gray-600 mb-2"
    >
      john@example.com
    </p>
    <footer
      class="flex justify-between items-center mt-4"
    >
      <p
        class="text-gray-600"
      >
        March 20, 2024
      </p>
      <p
        class="text-gray-600"
      >
        14:30
      </p>
    </footer>
  </div>
</div>
`;
